api/v1alpha1/databaseclusterrestore_types.go

```diff
@@ -144,3 +147,74 @@ type DatabaseClusterRestoreList struct {
 func init() {
 	SchemeBuilder.Register(&DatabaseClusterRestore{}, &DatabaseClusterRestoreList{})
 }
+
+// IsComplete indicates if the restoration process is complete (regardless successful or not).
+func (r *DatabaseClusterRestore) IsComplete(engineType EngineType) bool {
+	switch engineType {
+	case DatabaseEnginePXC:
+		return isPXCRestoreStatusComplete(pxcv1.BcpRestoreStates(r.Status.State))
+	case DatabaseEnginePSMDB:
+		return isPSMDBRestoreStatusComplete(psmdbv1.RestoreState(r.Status.State))
+	case DatabaseEnginePostgresql:
+		return isPGRestoreStatusComplete(pgv2.PGRestoreState(r.Status.State))
+	}
+	return true
+}
+
+func isPXCRestoreStatusComplete(status pxcv1.BcpRestoreStates) bool {
+	switch status {
+	case pxcv1.RestoreNew:
+		return false
+	case pxcv1.RestoreStarting:
+		return false
+	case pxcv1.RestoreStopCluster:
+		return false
+	case pxcv1.RestoreRestore:
+		return false
+	case pxcv1.RestoreStartCluster:
+		return false
+	case pxcv1.RestorePITR:
+		return false
+	case pxcv1.RestoreFailed:
+		return true
+	case pxcv1.RestoreSucceeded:
+		return true
+	}
+	return true
```

Comment:
I suggest we turn around all three functions to return `false` by default and only for selected statuses we return `true`.
Otherwise if a new status is added and we forget to update these functions, the new status will be considered as a "complete" status.

Reply:
That's a tough one :sweat_smile:

I'm leaning to agree with you @oksana-grishchenko.

If they add an end state and we have default `false` then there's no coming back.
In the other hand, if they add a transitory state and we say it's complete (default `true`), despite being wrong it's a transitory thing that will fix itself shortly.

Both case are wrong and we should really try to be in sync with the cloud team whenever they change this kind of stuff (I think that we do a good job of it with our weekly DBaaS sync). In the case this slips through, I think we should go with the option that fixes itself automatically, i.e. keeping `true` as a default.

Reply:
Both approaches have their pro and cons, but looks like we have a consensus regarding the topic. Got an approval from Michal to resolve the thread.